File: pergunta02.java
public class pergunta02 {

    public static void main(String[] args) {
        int informedNumber = 13;

        if (isFibonacci(informedNumber)) {
            System.out.println(informedNumber + " is present in Fibonacci");
        } else {
            System.out.println(informedNumber + " isn't present in Fibonacci");
        }
    }

    public static boolean isFibonacci(int w) {
        long x1 = 5L * w * w + 4;
        long x2 = 5L * w * w - 4;

        long x1Sqrt = (long) Math.sqrt(x1);
        long x2Sqrt = (long) Math.sqrt(x2);

        return (x1Sqrt * x1Sqrt == x1) || (x2Sqrt * x2Sqrt == x2);
    }
}

File: pergunta02.rs
fn main() {
    let informed_number = 13;

    if is_fibonacci(informed_number) {
        println!("{informed_number} is present in Fibonnaci");
    } else {
        println!("{informed_number} isn't present in Fibonnaci");
    }
}

fn is_fibonacci(w: i32) -> bool {
    let x1 = 5 * w.pow(2) + 4;
    let x2 = 5 * w.pow(2) - 4;

    let x1_sqrt = (x1 as f64).sqrt() as i64;
    let x2_sqrt = (x2 as f64).sqrt() as i64;

    x1_sqrt * x1_sqrt == x1 as i64 || x2_sqrt * x2_sqrt == x2 as i64
}
